tests/test_plugins.py

```diff
@@ -58,7 +60,7 @@ class Model(BaseModel, plugin_settings={'observe': 'all'}):
 
 def test_on_validate_json_on_error() -> None:
     class CustomOnValidateJson:
-        def enter(
+        def on_enter(
```

Comment:
None of these were getting called 😢. In my opinion, this is a strong argument for using a single method with a context manager instead of multiple optional callbacks.